elements/pl-xss-safe/pl-xss-safe.mustache

```diff
@@ -1,17 +1,20 @@
-<script data-code="{{contents}}" data-language="{{language}}">
-(function() {
-    let code = $(document.currentScript).data("code");
-    let language = $(document.currentScript).data("language");
+<script>
+  $(function() {
+    const element = $('#xss-{{uuid}}');
+    const language = element.attr("data-language");
+    let code = element.attr("data-code");
     // Uses the same JS library as the pl-file-editor for markdown
     // conversion and xss filter for compatibility and to avoid
     // confusion
     if (language == "markdown") {
-        let renderer = new showdown.Converter({
-            'literalMidWordUnderscores':true,
-            'literalMidWordAsterisks':true,
-        });
-        code = renderer.makeHtml(code);
+      let renderer = new showdown.Converter({
+        'literalMidWordUnderscores':true,
+        'literalMidWordAsterisks':true,
+      });
+      code = renderer.makeHtml(code);
     }
-    document.write(filterXSS(code));
-})();
+    code = filterXSS(code);
+    element.html(code);
+  });
 </script>
+<div id="xss-{{uuid}}" data-code="{{contents}}" data-language="{{language}}"></div>
```

Comment:
This was changed because `document.write` behaves erratically when loaded dynamically. Also, `document.currentScript` can have some weird behaviour when loaded dynamically. That said, from a quick review it seems `pl-drawing` also uses `document.currentScript`, but I have no experience with the element to be able to test it properly, so if someone else can test it that would be great.